src/zones/dbm.rs

```diff
@@ -676,6 +676,79 @@ impl DBM<Valid> {
         unsafe { dbm.assert_valid() }
     }
 
+    fn compute_tables(&self, src_clocks: &Vec<bool>, dst_clocks: &Vec<bool>) -> (Vec<ClockIndex>, Vec<ClockIndex>) {
+        let mut dst_to_src = Vec::<ClockIndex>::default();
+        let mut src_to_dst = Vec::<ClockIndex>::default();
+        let mut src_ind = 0;
+        assert_eq!(src_clocks.len(), dst_clocks.len());
+
+        for (src, dst) in src_clocks.iter().zip(dst_clocks.iter()){
+            let (src, dst) = (src.to_owned(), dst.to_owned());
+            if dst {
+                src_to_dst.push(dst_to_src.len());
+                if src {
+                    dst_to_src.push(src_ind);
+                }
+                else {
+                    dst_to_src.push(0);
+                }
+            }
+            else {
+                src_to_dst.push(0);
+            }
+
+            src_ind += 1;
+        };
+
+        return (src_to_dst, dst_to_src);
+    }
+
+    fn update_dbm(&mut self, src: &DBM<Valid>, dst_to_src: &Vec<ClockIndex>) {
+        self.data[0] = LE_ZERO;
+
+        for j in 1..self.dim {
+            if dst_to_src[j] == 0
+            {
+                self.data[j] = LE_ZERO;
+            }
+            else
+            {
+                self.data[j] = src.data[dst_to_src[j]];
+            }
+        }
+
+        for i in 1..self.dim {
+            if dst_to_src[i] != 0 { // If copy from src.
+                let constraint0 = src.data[src.dim * dst_to_src[i]];
+                self.data[i * self.dim] = constraint0;
+
+                for j in 1..self.dim {
+                    if dst_to_src[j] == 0
+                    {
+                        self.data[j + i * self.dim] = constraint0;
+                    }
+                    else {
+                        self.data[j + i * self.dim] = src.data[dst_to_src[j] + src.dim * dst_to_src[i]];
+                    }
+                }
+            }
+            else { // Insert new row.
+                for j in 0..self.dim {
+                    self.data[j + i * self.dim] = LS_INFINITY;
+                }
+            }
+            self.data[i + i * self.dim] = LE_ZERO;
+        }
+    }
+
+    fn shrink_expand(&self, src_clocks: &Vec<bool>, dst_clocks: &Vec<bool>) -> (DBM<Valid>, Vec<ClockIndex>) {
+        assert_eq!(src_clocks.len(), dst_clocks.len());
+        let (src_to_dst, dst_to_src) = self.compute_tables(src_clocks, dst_clocks);

```

Comment:
I don't think it is that efficient to calculate remove_clock([2,3]) to src_clocks([T, T, F, F, T]) to then update_dbm([0, 1, 4])